### DISCUSSION TITLE
dprepro not working in python script

### DISCUSSION THREAD

**Discussion by kwesi-boafo** [snl-dakota/dakota#20] [Q&A] [answered]
Hello Community,
I am running a python script and my call to 'dprepro' is not successful. The line in the script is shown below:
os.system('dprepro' + sys.argv[1] + 'Gh.template Gh.inp')

Thanks in advance for your help.

/Emmanuel

**Comment by jadamstephens** [comment, marked as answer]
The command provided to `os.system()` needs some whitespace: `os.system('dprepro ' + sys.argv[1] + ' Gh.template Gh.inp')`.

However, since your driver is written in Python, you can also use the `dakota.interfacing` Python module to run dprepro. Here's the [documentation](https://dakota.sandia.gov/sites/default/files/docs/6.17.0-release/user-html/usingdakota/inputfile/interfaces/dakotainterfacing.html#processing-templates).

**Reply by kwesi-boafo** [reply]
Hello Adam,
Thank you for the suggestion, dprepro now executes as expected. The hurdle I am facing now is to execute the solver with a python script. I have tried a few options but without success so far. The solver executable takes the name of the input file and 'CR' in order to run. I created a file with this information and expects the executable to read from this file ('in'). Below are the options I have used so far:
subprocess.Popen('EXECUTABLE in', shell=True)
os.system(r"EXECUTABLE Gh CR")

In bash, (EXECUTABLE < in) works.

Best regards

**Reply by jadamstephens** [reply]
In Bash, the `<` means input redirection. The file `in` is being read by Bash and sent to `EXECUTABLE` line by line on stdin. It's as though a user is sitting at the command line and typing in what he sees in `in`, hitting return after each line.

The two Python examples don't have that behavior. There, the name of the file `in` is being provided as a command line argument. It assumes that `EXECUTABLE` will open the file `in` and read it itself.

Your Python examples probably will work if you change the string to `'EXECUTABLE < in'`, as you have it in your Bash script.

**Reply by kwesi-boafo** [reply]
Yes, it works as you correctly suggested.
I appreciate your time Adam!
